lib/avo/fields/base_field.rb

```diff
@@ -99,14 +99,20 @@ def initialize(id, **args, &block)
         except_on args[:except_on] if args[:except_on].present?
       end
 
-      def hydrate(model: nil, resource: nil, action: nil, view: nil, panel_name: nil, user: nil)
-        @model = model if model.present?
-        @view = view if view.present?
-        @resource = resource if resource.present?
-        @action = action if action.present?
-        @user = user if user.present?
-        @panel_name = panel_name if panel_name.present?
+      def hydrate(**kwargs)
+        # List of permitted keyword argument keys as symbols
+        permited_kwargs_keys = %i[model resource action view panel_name user]
 
+        # Check for unrecognized keys
+        unrecognized_keys = kwargs.keys - permited_kwargs_keys
+        raise ArgumentError, "Unrecognized argument(s): #{unrecognized_keys.join(', ')}" if unrecognized_keys.any?
+
+        # Set instance variables with provided values
+        kwargs.each do |key, value|
+          instance_variable_set("@#{key}", value)
+        end
```

Comment:
why the refactor here? it feels difficult to understand at a glance

Reply:
The existing code made it impossible to hydrate the field with `nil` values, resulting in undesired behavior. For instance, in cases where a field within an action relied on the `@record` via a default Proc, it would correctly receive the default value when executed in a show view. However, when executed in an index view, the field attempted to hydrate with `model: nil`, yet the prior `model` would remain hydrated, causing the value to be fetched incorrectly.

Reply:
What this does is if we use `hydrate(model: nil)` will assign `@model = nil` without touching the other variables... With the old code `@model = model if model.present?` since `@model.present? == false` the `@model` will keep the last assigned `model` even if we want to force hydrate it to nil...